scheduler/graphql_mid/schema.py

```diff
@@ -23,15 +24,19 @@ async def new_schedule(self,
         try:
             start, end = Time(new_schedule_input.start_time, format='iso', scale='utc'), \
                          Time(new_schedule_input.end_time, format='iso', scale='utc')
-            scheduler = build_scheduler(start, end)
+            scheduler = build_scheduler(start, end, new_schedule_input.site)
         except ValueError:
             # TODO: log this error
             return NewScheduleError(error='Invalid time format. Must be ISO8601.')
         else:
-            pm = setup_manager()
-            pm.add_task(datetime.now(), scheduler, TaskType.STANDARD)
-            # await asyncio.sleep(10) # Use if you want to wait for the scheduler to finish
-            return NewScheduleSuccess(success=True)
+            try:
+                pm = setup_manager()
+                pm.add_task(datetime.now(), scheduler, TaskType.STANDARD)
+                await asyncio.sleep(10) # Use if you want to wait for the scheduler to finish
```

Comment:
Seems like kind of an arbitrary number? Why 10 s?

Reply:
It is. Is a hack to force the waiting so I can test that in the UI. It should be removed in further changes